File: behavioral/dart/chain_of_responsibility_structured/loggers.dart
import 'interfaces.dart';

class ConsoleLogger extends Logger {
  @override
  void log(String message, int severity) {
    if (severity >= 1) {
      print('[CONSOLE] $message');
    }
    passLog(message, severity);
  }
}

class FileLogger extends Logger {
  @override
  void log(String message, int severity) {
    if (severity >= 2) {
      print('[FILE] $message');
    }
    passLog(message, severity);
  }
}

class ErrorLogger extends Logger {
  @override
  void log(String message, int severity) {
    if (severity >= 3) {
      print('[ERROR] $message');
    }
    passLog(message, severity);
  }
}

File: behavioral/typescript/chain_of_responsibility_structured/loggers.ts
import { Logger } from './interfaces';

export class ConsoleLogger extends Logger {
  log(message: string, severity: number): void {
    if (severity >= 1) {
      console.log(`[CONSOLE] ${message}`);
    }
    this.passLog(message, severity);
  }
}

export class FileLogger extends Logger {
  log(message: string, severity: number): void {
    if (severity >= 2) {
      console.log(`[FILE] ${message}`);
    }
    this.passLog(message, severity);
  }
}

export class ErrorLogger extends Logger {
  log(message: string, severity: number): void {
    if (severity >= 3) {
      console.log(`[ERROR] ${message}`);
    }
    this.passLog(message, severity);
  }
}
